### PR TITLE
Add `google-cloud-cpp-ai` feedstock

### PR DESCRIPTION
**Pull request by coryan**
This feedstock will contain the AI and ML related features in `google-cloud-cpp`. Some of the largest features in `google-cloud-cpp` fall in this category, and the overall count of libraries is manageable.

Splitting these features to a separate feedstock will reduce the build time for the `google-cloud-cpp` feedstock, and should keep it below the CI timeouts in the medium term.

Part of the work for https://github.com/conda-forge/google-cloud-cpp-feedstock/issues/141

- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/5eddbd7fc9d1502169089da06c3688d9759be978/recipes/example/meta.yaml#L64-L73) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Package does not ship static libraries. If static libraries are needed, [follow CFEP-18](https://github.com/conda-forge/cfep/blob/main/cfep-18.md).
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

### CODE REVIEW

**Comment by coryan** [comment]
@h-vetinari this looks ready for review.  The failure on Windows looks like a transient:

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=848229&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=7c0f8eae-6d6f-51bf-636f-73a1a7fb1bc4&l=11291

```
ERROR conda.core.link:_execute(945): An error occurred while installing package 'conda-forge::vc14_runtime-14.38.33130-h82b7239_18'.
... ... ...
... ...
conda.CondaMultiError: [Errno 13] Permission denied: 'C:\\bld\\google-cloud-cpp-ai_1703168478740\\_h_env\\vcruntime140.dll'
```

Thoughts?

**Comment by coryan** [comment]
@conda-forge/help-c-cpp I do not understand why this is failing on Windows.  Any suggestions on how to troubleshoot the problem?

**Comment by h-vetinari** [comment]
The failures is directly caused by the failure further up to correctly tear down the previous environment:
```
WARNING conda.gateways.disk.delete:unlink_or_rename_to_trash(188): Could not remove or rename C:\bld\google-cloud-cpp-ai_1704469478201\_h_env\vcruntime140.dll.conda_trash.conda_trash.conda_trash.conda_trash.conda_trash.conda_trash.conda_trash.conda_trash.conda_trash.conda_trash.conda_trash.conda_trash.conda_trash.conda_trash.conda_trash.conda_trash.conda_trash.conda_trash.conda_trash.  Please remove this file manually (you may need to reboot to free file handles)
```

That's normally a transient thing, but it seems you're able to reproduce this? It's also particular that several outputs are packaged correctly before this happens.

**Comment by coryan** [comment]
> 
> That's normally a transient thing, but it seems you're able to reproduce this?

I have ran the build (only) twice.  Both times it seems to fail at the same spot:

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=848229&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=7c0f8eae-6d6f-51bf-636f-73a1a7fb1bc4&l=11230

https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=853346&view=logs&j=240f1fee-52bc-5498-a14a-8361bde76ba0&t=7c0f8eae-6d6f-51bf-636f-73a1a7fb1bc4&l=11166

I would call that a repro, though it is a bit odd.

> It's also particular that several outputs are packaged correctly before this happens.

Indeed.  I wonder if it is specific to `texttospeech-devel`, or maybe the number of outputs packaged before `texttospeech`.  I am going to (temporarily) remove `texttospeech*` and see what the results are.

**Comment by coryan** [comment]
Note to self: after removing `texttospeech*` it fails in `translate*`.  I will try removing more libraries and see if that fixes the build problems.

**Comment by coryan** [comment]
@h-vetinari I think this is ready for review.  I paid attention to the feedstock name, avoided leading `.build` names, and I think I got the `host`, `build` and `run_exports` right.

File: recipes/google-cloud-cpp-ai/bld.bat [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+
+FOR %%G IN (
+  aiplatform
+  automl
+  discoveryengine
+  dialogflow_es
+  dialogflow_cx
+  dlp
+  speech) DO (
+    cmake -G "Ninja" ^
```
Comment (h-vetinari) [inline recipes/google-cloud-cpp-ai/bld.bat:17]: Nit: readability.
```suggestion
  speech
) DO (
    cmake -G "Ninja" ^
```

**Review by h-vetinari — APPROVED** [review]
This looks very good (getting easier by repetition for both of us ;-)), thank you!

The nit is not worth a further holdup here, I suggest you fix it on the feedstock if you care.

I checked the logs, very nice to see warning-free link checks (though on windows I'm looking forward to the switch to shared builds! ;-))

**Pull request merged**